modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/programs/Genesis.scala

```diff
@@ -20,15 +21,18 @@ import org.tessellation.sdk.domain.collateral.{Collateral, OwnCollateralNotSatis
 import org.tessellation.sdk.domain.snapshot.storage.SnapshotStorage
 import org.tessellation.sdk.http.p2p.clients.StateChannelSnapshotClient
 import org.tessellation.sdk.infrastructure.consensus.ConsensusManager
+import org.tessellation.sdk.infrastructure.genesis.{Loader => GenesisLoader}
 import org.tessellation.security.SecurityProvider
 
+import fs2.io.file.Path
 import org.typelevel.log4cats.slf4j.Slf4jLogger
 
-trait GenesisService[F[_]] {
+trait Genesis[F[_]] {
+  def accept(path: Path): F[Unit]
   def accept(genesis: CurrencySnapshot): F[Unit]
 }
 
-object GenesisService {
+object Genesis {
```

Comment:
both `Service` and `Program` are part of a domain therefore should have no dependency to infrastructure. changing `GenesisService` into `GenesisProgram` makes no difference. both previous and current code has a dependency do infrastructure. if there is an easy way to fix that, please check it out. otherwise we can think about bigger refactor out of scope of this change
![image](https://user-images.githubusercontent.com/4959931/233018525-8b33a685-7eb8-4636-8995-d3b0eab560f6.png)

Reply:
I agree, but unfortunately, it is a part of more extensive refactoring. Changing Service to Program doesn't fix the dependency of course but at least points out that Genesis and Rollback should be programs. We need to plan a bigger refactoring because getting rid of infrastructure dependencies there is not that trivial. I have it fixed on a different change so you will see if you are fine with the consequences of making these programs infra-independent.